taca/analysis/analysis_nanopore.py

```diff
@@ -1,462 +1,295 @@
-    """
-    try:
-        parse(string, fuzzy=False)
-        return True
-    except ValueError:
-        return False
+def find_run_dirs(dir_to_search: str, skip_dirs: list):
+    """Takes an input dir, expected to contain ONT run dirs.
+    Append all found ONT run dirs to a list and return it"""
 
+    logger.info(f"Looking for ONT runs in {dir_to_search}...")
 
-def find_minion_runs(minion_data_dir, skip_dirs):
-    """Find nanopore runs to process."""
     found_run_dirs = []
-    try:
-        found_top_dirs = [
-            os.path.join(minion_data_dir, top_dir)
-            for top_dir in os.listdir(minion_data_dir)
-            if os.path.isdir(os.path.join(minion_data_dir, top_dir))
-            and top_dir not in skip_dirs
-        ]
-    except OSError:
-        logger.warning(
-            "There was an issue locating the following directory: {}. "
-            "Please check that it exists and try again.".format(minion_data_dir)
-        )
-    # Get the actual location of the run directories in /var/lib/MinKnow/data/QC_runs/USERDETERMINEDNAME/USERDETSAMPLENAME/run
-    if found_top_dirs:
-        for top_dir in found_top_dirs:
-            if os.path.isdir(top_dir):
-                for sample_dir in os.listdir(top_dir):
-                    if os.path.isdir(os.path.join(top_dir, sample_dir)):
-                        for run_dir in os.listdir(os.path.join(top_dir, sample_dir)):
-                            found_run_dirs.append(
-                                os.path.join(top_dir, sample_dir, run_dir)
-                            )
-    else:
-        logger.warning(
-            "Could not find any run directories in {}".format(minion_data_dir)
-        )
+    for found_dir in os.listdir(dir_to_search):
+        if (
+            os.path.isdir(os.path.join(dir_to_search, found_dir))
+            and found_dir not in skip_dirs
+            and re.match(ONT_RUN_PATTERN, found_dir)
+        ):
+            logger.info(f"Found ONT run {found_dir} in {dir_to_search}")
+            found_run_dirs.append(os.path.join(dir_to_search, found_dir))
+
     return found_run_dirs
 
 
-def find_ont_transfer_runs(ont_data_dir, skip_dirs):
-    """Find runs in ngi-nas.
-    These are assumed to be flowcell dirs, not project dirs.
-    """
-    try:
-        found_dirs = [
-            os.path.join(ont_data_dir, top_dir)
-            for top_dir in os.listdir(ont_data_dir)
-            if os.path.isdir(os.path.join(ont_data_dir, top_dir))
-            and top_dir not in skip_dirs
-        ]
-    except OSError:
-        logger.warning(
-            "There was an issue locating the following directory: {}. "
-            "Please check that it exists and try again.".format(ont_data_dir)
-        )
-    return found_dirs
+def send_error_mail(run_name, error: BaseException):
 
+    email_subject = f"Run processed with errors: {run_name}"
+    email_message = "{}\n\n{}".format(
```

Comment:
Why not use an f-string here too?

Reply:
[8f78c86](https://github.com/SciLifeLab/TACA/pull/386/commits/8f78c865c7f28cb945c221f135054a9e8476207e)